### PR TITLE
Separate TUF and trusted root management code

### PR DESCRIPTION
**Pull request by jku**
#### Summary

The purpose of this is to later enable both "--trust-root <FILE>" and some sort of "--offline" functionality.
 * Trust root can now be initialized from tuf, offline tuf or from a file
 * `_internal.tuf` module is now used only from the new `_internal.trustroot` module
 * all "trust root parsing" was moved from `_internal.tuf` to `_internal.trustroot`
 * Tests are modified to use the trustroot module now but they still (also) test the internal TUF implementation details
 * The new functionality (offline & from_file) is tested but is not exposed to UI

#### Some details

The main functional change as I see it is in TrustedRoot/TrustUpdater initialization. There are three possible scenarios (once this is exposed through UI):
* A trust root is provided as a file: TrustedRoot never instantiates a TrustUpdater
* offline is requested: A TrustUpdater is instantiated but it does no TUF operations (only returns trust root from cache)
* default: TrustUpdater is instantiated and it refreshes from repository immediately (since we know it will be used)

This leads to some minor test changes but nothing too bad.

#### Release Note

No release notes, I think: This should not change API or cli at this point

### CODE REVIEW

File: sigstore/_internal/trustroot.py
Comment (woodruffw) [inline sigstore/_internal/trustroot.py:59]: My 0.02c: I'm a fan of the "import as private, export same name" idiom:

```python
from whereever import TrustedRoot as _TrustedRoot

class TrustedRoot(_TrustedRoot):
    ...
```

This avoids a naming decision and is still unambiguous, since Python types convey their namespace 🙂

**Comment by woodruffw** [comment]
Reviewing now!

@jku: mind doing an update/rebase? I'd do it myself, but that then causes problems with the number of required approvers.

File: sigstore/_cli.py
Comment (jku) [inline sigstore/_cli.py:662]: there's a slight change of functionality here (in contrast to what I said in the PR message):
this now leads to TUF repo getting  refreshed even if all keys are provided via command line... I'll think about how to avoid that

Comment (woodruffw) [reply sigstore/_cli.py:662]: Making sure I understand: this causes the TUF repo to get refreshed because `TrustedRoot.production()` is doing the refresh, rather than deferring it to one of the `get_*_keys()` calls, right?

Comment (jku) [reply sigstore/_cli.py:662]: Yes. In the future (if the local keys can only be provided via --trust-root) this would not be an issue but in this transition state it could be

Comment (woodruffw) [reply sigstore/_cli.py:662]: Makes sense, thanks. We can hold off if you feel strongly here, but I'm personally okay with this -- we intend to remove these flags anyways, so IMHO we can deal with a temporary unnecessary network request.

**Review by woodruffw — APPROVED** [review]
LGTM! 

@jku feel free to merge unless you'd like to address https://github.com/sigstore/sigstore-python/pull/844#discussion_r1440622945 in this PR; I'm fine either way 🙂

File: sigstore/_cli.py
Comment (jku) [reply sigstore/_cli.py:662]: Let's leave it like this then and move on with next steps

**Pull request merged**
